apps/crn-server/test/routes/research-outputs.route.test.ts

```diff
@@ -86,6 +86,79 @@ describe('/research-outputs/ route', () => {
         expect(response.status).toBe(400);
       });
     });
+
+    describe('Drafts', () => {
+      test('Should only return drafts', async () => {
+        const listResearchOutputResponse = getListResearchOutputResponse({
+          published: false,
+        });
+
+        researchOutputControllerMock.fetch.mockResolvedValueOnce(
+          listResearchOutputResponse,
+        );
+
+        const response = await supertest(app).get('/research-outputs');
+
+        expect(response.body.items[0].published).toBe(false);
+      });
+
+      test('Should return a validation error when additional fields exist', async () => {
+        const response = await supertest(app).get(`/research-outputs`).query({
+          status: 'draft',
+          teamId: '123',
+          additionalField: 'some-data',
+        });
+
+        expect(response.status).toBe(400);
+      });
+
+      test('Should call the controller with the right parameters', async () => {
+        await supertest(app).get('/research-outputs').query({
+          take: 15,
+          skip: 5,
+          status: 'draft',
+          teamId: 'team-id-0',
+          workingGroupId: 'workingGroup-id-0',
+        });
+
+        const expectedParams = {
+          take: 15,
+          skip: 5,
+          filter: {
+            teamId: 'team-id-0',
+            status: 'draft',
+            workingGroupId: 'workingGroup-id-0',
+          },
+          includeDrafts: true,
+        };
+
+        expect(researchOutputControllerMock.fetch).toBeCalledWith(
+          expectedParams,
+        );
+      });
+
+      test("Should return 403 if user is not associated to output's team", async () => {
+        const response = await supertest(app).get('/research-outputs').query({
+          take: 15,
+          skip: 5,
+          status: 'draft',
+          teamId: 'team-id-403',
+        });
+
+        expect(response.status).toBe(403);
+      });
+
+      test("Should return 403 if user is not associated to output's working group", async () => {
+        const response = await supertest(app).get('/research-outputs').query({
+          take: 15,
+          skip: 5,
+          status: 'draft',
+          workingGroupId: 'workingGroup-id-403',
```

Comment:
to be fair both this and the test above should also contain a fixture for the loggedInUser - in this file we don't actually have a mock function for the currently logged user but there is a way. We can keep it as is for now but I know @oratiu is working on a similar piece of work where he is adding such a fixture, might be a good time to then add them here too

Reply:
do we have a JIRA ticket about this so i can add this file to?